### ISSUE TITLE
node_modules ignored

### ISSUE THREAD

**Issue by wayneseymour** [olov/ng-annotate#178]
`ng-annotate/.gitignore` has an entry for `node_modules`.

This is very bad, please remove.

Have a good day.

**Comment by johnsonw** [comment]
+1

**Comment by jgrund** [comment]
Having the same issue here. Trying to check in this module to source_control with it's deps. This .gitignore makes doing so difficult.

**Comment by olov** [comment]
Thanks for filing the issue - fixed in master

**Comment by mzgol** [comment]
I don't really understand the issue. For me the new situation is bad, just building the project makes me generate a lot of files that are neither ignored nor tracked by git. This makes git slow, makes it harder to switch between branches and makes it harder to work with git in general.

No project I know that doesn't keep its `node_modules` in source control. It's either included or ignored.

If someone really wants to check in the ng-annotate repository together with all generated files, it's easy - just `git add ng-annotate --force`.

**Comment by mzgol** [comment]
I've actually wanted to report an issue going in the opposite direction - the built final files are also not ignored and their placeholders are in the repo so building the binaries makes untracked as well as tracked & modified files to be generated.

I think a good strategy is to have `.gitignore` set up in a way that installing dependencies & building the project doesn't leave your repo in a dirty state. :/

**Comment by jgrund** [comment]
@mzgol Can you not add the files you don't want to track to an upstream .gitignore? 

With a downstream .gitignore you can no longer override including / excluding files.

Further, I could see usefulness of having a .gitignore for source control, but it should be stripped out before publishing to npm.

**Comment by jgrund** [comment]
@olov Any chance to publish a new npm release on this?

**Comment by mzgol** [comment]
> @mzgol Can you not add the files you don't want to track to an upstream .gitignore?

What do you mean about an upstream `.gitignore`? I mean that if I ever want to submit a PR to this repository, do anything here with the code, after 50cff455236fdf3748a718f99ac39c301cf3f0ad it's going to be extremely painful. Also, no other popular repository does this.

I think this change was a mistake.

**Comment by wayneseymour** [comment]
A mistake? Explain?

**Comment by mzgol** [comment]
> @olov Any chance to publish a new npm release on this?

Wait, `.gitignore` is not published, `.npmignore` is. I thought you wanted to not have the `.gitignore` in the repo, not in the published package. After npm installing ng-annotate I have this in `node_modules/ng-annotate` directory:
```
.npmignore
BUILD.md
CHANGES.md
LICENSE
README.md
build
defs-config.json
generate-sourcemap.js
lut.js
ng-annotate-harmony
ng-annotate-main.js
ng-annotate.js
nginject.js
node_modules
optionals
package.json
plugin-example.js
pos-to-linecolumn.js
run-tests.js
scope.js
scopetools.js
tests
```

No `.gitignore`.

I suppose npm is renaming `.gitignore` to `.npmignore` if the latter doesn't exist. If you're installing the package, you shouldn't have `.gitignore` in the directory. What npm version do you use?

BTW, if an older npm doesn't do the renaming, one solution would be to revert the `.gitignore` change but create `.npmignore` with contents:
```
.idea
.gitignore
```
You then won't have `.gitignore` in the published package but it will be present in the repository, removing both your & my problems.

@wayneseymour 
> A mistake? Explain?

I already did: https://github.com/olov/ng-annotate/issues/178#issuecomment-114924824

**Comment by jgrund** [comment]
Yes, I can see that. My issue is this .gitignore was making into the published module, which was causing us pain trying to integrate in our project. 

The correct fix is to keep .gitignore out of the published module.

**Comment by omsmith** [comment]
Perhaps .gitignore just shouldn't be included in the package.

Think the proper fix here is to add _.gitignore_ to _.npmignore_

**Comment by mzgol** [comment]
> The correct fix is to keep .gitignore out of the published module.

+1, that I have nothing against.

**Comment by olov** [comment]
Thanks for the insightful comments. I'm surprised `.gitignore` was included in the npm package in the first place. Now fixed in https://github.com/olov/ng-annotate/commit/35261777fad14a421e26310d5e7cf2283c7d7df7. I'll push out a patch release if everyone's ok with this resolution.

**Comment by mzgol** [comment]
Latest master is OK for me. :+1:

**Comment by jgrund** [comment]
Looks good here, thanks for quick turnaround.

**Comment by wayneseymour** [comment]
Thanks man.  I'm ok with the resolution.  Please ship this.  :)

**Comment by olov** [comment]
released in 1.0.1

**Comment by wayneseymour** [comment]
Woohoo
On Wed, Jun 24, 2015 at 6:04 PM Olov Lassus <notifications@github.com>
wrote:

> released in 1.0.1
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/olov/ng-annotate/issues/178#issuecomment-115024707>.
>
-- 
W. K. Seymour III
Software Engineer
Intel Corporation
High Performance Data Division
